cfg/types.go

```diff
@@ -114,3 +114,29 @@ func (p *ResolvedPath) UnmarshalText(text []byte) error {
 	*p = ResolvedPath(path)
 	return nil
 }
+
+// OptimizationInput provides runtime context for applying optimizations.
+// This struct can be extended in the future to support additional optimization
+// dimensions such as region, storage class, or project-specific settings.
+type OptimizationInput struct {
+	// BucketType specifies the GCS bucket type.
+	// An empty string means no bucket-type-based optimization should be applied.
+	BucketType BucketType
+}
+
+// BucketType represents the type of GCS bucket.
+type BucketType string
+
+const (
+	// BucketTypeZonal represents a zonal bucket with single-zone storage.
+	BucketTypeZonal BucketType = "zonal"
+
+	// BucketTypeHierarchical represents a bucket with hierarchical namespace enabled.
+	BucketTypeHierarchical BucketType = "hierarchical"
+
+	// BucketTypeStandard represents a standard (regional or multi-regional) bucket.
+	BucketTypeStandard BucketType = "standard"
```

Comment:
Opposite of hierarchical and as we usually call this bucket-type

```suggestion
	BucketTypeFlat BucketType = "flat"
```